tests/src/com.github.ben-manes.caffeine/caffeine/2.9.3/src/test/java/com_github_ben_manes_caffeine/caffeine/EvictionTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_github_ben_manes_caffeine.caffeine;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Expiry;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import org.junit.jupiter.api.Test;
+
+import java.time.ZonedDateTime;
+import java.util.concurrent.TimeUnit;
+
+import static java.time.temporal.ChronoUnit.MILLIS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class EvictionTest {
+    @Test
+    void testSizeBased() {
+        LoadingCache<String, String> firstGraphs = Caffeine.newBuilder().maximumSize(10_000).build(key -> key.equals("Hello") ? "World" : "Universe");
+        assertThat(firstGraphs.get("Hello")).isEqualTo("World");
```

Comment:
Isn't this `firstGraphs` testing same as in [testRefresh](https://github.com/oracle/graalvm-reachability-metadata/pull/388/files#diff-220cd86ae77195bc08dc7249346abd6ccbbbd4ac5f7b828f0bd266b128cc25baR27) test? The only difference is additional `refreshAfterWrite` function call in the first test. Therefore this `firstGraphs` part of this test can be removed. Is that correct?

Reply:
- There is a problem with the design of the `testRefresh()` function. The relevant issue record is not in a git, but in my personal blog. When writing relevant unit tests, the unit test for the Refresh function used Mockito to mock `java.util.concurrent.Executor`, and the relevant unit tests failed under GraalVM Native Image due to the use of Mockito's inline plugin. Therefore, when a PR was refactored, this unit test removed the use of the Mockito Java API, making the related unit tests look strange.
- I refactored the `testRefresh()` function to accurately assert actual changes in the Refresh functionality. Apparently this also introduces some extra JSON for `com.github.benmanes.caffeine.cache.SSR`.
- Not sure what I'm missing, but it looks like the version of gradle wrapper currently used by git doesn't support graalvm ce for jdk21.